src/generic_api.ts

```diff
@@ -0,0 +1,51 @@
+import querystring = require('querystring');
+import pluralize = require('pluralize');
+
+export interface K8sObject {
+    apiVersion: string;
+    kind: string;
+    metadata?: {
+        name?: string;
+        namespace?: string;
+        labels?: object;
+    };
+}
+
+export function CreatePath(object: K8sObject, include_name: boolean = true) {
+  if (!object.apiVersion) {
+    throw new Error('The object passed must contain apiVersion field')
+  }
+  if (!object.kind) {
+    throw new Error('The object passed must contain kind field')
+  }
+
+  let path = ""
+
+  // add apiVersion
+  if (object.apiVersion == "v1") {
+    path += "/api/v1"
+  } else {
+    path += "/apis/" + object.apiVersion
+  }
+
+  // add namespace if object namespaced
+  if (object.metadata?.namespace) {
+    path += "/namespaces/" + object.metadata.namespace
+  }
+
+  // add object kind
+  path += "/" + pluralize(object.kind.toLowerCase())
```

Comment:
fwiw, the plural  information is available via the Discovery API on the Kubernetes cluster. This probably works for nearly everything, so it's an ok approach for now, but perhaps add:

`// TODO: Use discovery api to get the correct plural`